src/Pages/Applications/DevtronApps/Details/AppConfigurations/Navigation/EnvConfigurationsNav.tsx

```diff
@@ -306,15 +307,23 @@ export const EnvConfigurationsNav = ({
     compareWithURL,
     appOrEnvIdToResourceApprovalConfigurationMap,
     isTemplateView,
+    shouldSetEnvInContext = false,
 }: EnvConfigurationsNavProps) => {
     const history = useHistory()
     const { isSuperAdmin } = useMainContext()
+    const { setEnvironmentId } = useAppContext()
     const { pathname } = useLocation()
     const { path, params } = useRouteMatch<EnvConfigRouteParams>()
 
     const { envId, resourceType } = params
     const parsedResourceId = +params[paramToCheck]
 
+    useEffect(() => {
+        if (envId && !Number.isNaN(+envId) && shouldSetEnvInContext) {
```

Comment:
```suggestion
        if (shouldSetEnvInContext && envId && !Number.isNaN(+envId)) {
```

and add `shouldSetEnvInContext` in dependency array